sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthFileCredential.java

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.identity.implementation.IdentityClientOptions;
+import reactor.core.publisher.Mono;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.InvalidPathException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Enables authentication to Azure Active Directory using configuration
+ * information stored Azure SDK Auth File.
+ *
+ * An Azaure SDK Auth file may be generated by passing <code>--sdk-auth</code>
+ * when generating an service principal with
+ * <code>az ad sp create-for-rbac</code>. At this time
+ * <see cref="AuthFileCredential"/> only supports SDK Auth Files which contain a
+ * client secret, client certificates are not supported at this time.
+ */
+@Immutable
+public class AuthFileCredential implements TokenCredential {
+    private final static ClientLogger logger = new ClientLogger(AuthFileCredential.class);
+    private IdentityClientOptions identityClientOptions;
+    TokenCredential credential;
+    private static final SerializerAdapter SERIALIZER_ADAPTER = JacksonAdapter.createDefaultSerializerAdapter();
+
+    /**
+     * Creates an instance of the SdkAuthFileCredential class based on information
+     * in given SDK Auth file. If the file is not found or there are errors parsing
+     * it, <see cref="GetToken(TokenRequestContext)"/> and
+     * <see cref="GetToken(TokenRequestContext)"/> will throw a
+     * <see cref="Exception"/> with details on why the file could not be used.
+     *
+     * @param filepath              The path to the SDK Auth file.
+     * @param identityClientOptions the options for configuring the identity client
+     */
+    AuthFileCredential(String filepath, IdentityClientOptions identityClientOptions) {
+        Objects.requireNonNull(filepath, "'filepath' cannot be null.");
+        Objects.requireNonNull(identityClientOptions, "'identityClientOptions' cannot be null.");
+        this.identityClientOptions = identityClientOptions;
+        if (credential == null) {
+            try {
+                credential = BuildCredentialForCredentialsFile(ParseCredentialsFile(filepath));
```

Comment:
Since this performs IO I think this should be done in getToken rather than in the constructor.

Reply:
Thanks @schaabs for clarifying. I previously asked @Luyunmt  to fail earlier by checking the credentials file earlier. @Luyunmt I think we would still want to throw in `getToken` instead of the constructor.